[ADD] Sign: section on Field Types (#1050)
odoo/documentation: content/applications/finance/sign.rst

====
Sign

> ====

Reviewer: Proposal of improvement (open debate):
It might be interesting to specify in the beginning of the documentation of enterprise features that they are restricted to enterprise users.
It's already done in the developer doc to specify views restricted to enterprise (https://www.odoo.com/documentation/15.0/developer/reference/backend/views.html?highlight=views#cohort), but I believe it might make sense in the user doc as well.

cc: @jcs-odoo 

If it's ok for you, you can integrate the tag with the following rst code:
```rst

.. raw:: html

   <span class="badge" style="background-color:#AD5E99">Enterprise feature</span>
```

Reviewer 2: This is indeed a good suggestion, but the user doc only focuses on Enterprise for now.
I'm worried that if we add this here for this doc only, it may mislead readers when they read other pages.

I'll keep this in mind for the future, but I wouldn't haste it in this PR right now.

Thanks

.. seealso::
   - `Odoo Sign: product page <https://www.odoo.com/app/sign>`_

>    - `Odoo Tutorials: Sign <https://www.odoo.com/slides/sign-61>`_

Reviewer 2: only one version: the latest, or even unreleased :D 
At least, theoretically.
Let's just say that it's only one link for all versions ^^

   informational purposes only. As laws governing electronic signatures rapidly evolve, we cannot
   guarantee whether all information is up to date or not. We advise contacting a local attorney for
   legal advice regarding electronic signature compliance and validity.

.. _sign/field-types:

Field Types
===========

By configuring your own *Field Types*, also known as *Signature Item Types*, you can make the
signing process even faster for your customers, partners, and employees.

To create and customize fields, activate the :ref:`developer mode <developer-mode>`. Then head to
:menuselection:`Documents --> Configuration --> Field Types` and click on *Create*.

After giving your new field a name, select one of the six *Types* available:

- **Signature**: users are prompted to enter their signature either by drawing it, automatically
  generating one based on their name, or uploading a local file (usually an image). Each subsequent
  *Signature* field then reuses the data entered in the first field.
- **Initial**: users are prompted to enter their initials, in a similar way to the *Signature*
  field.
- **Text**: users enter text on a single line.

> - **Multiline Text**: user enter text on multiple lines.

Author: Thanks for spotting this inconsistency!

   The information provided here does not constitute legal advice; it is provided for general

>    informational purposes only. As laws governing electronic signatures evolve rapidly, we cannot

Author: Note that we usually don't comment on parts that weren't changed - unless the changes/elements introduced have an impact on said parts. I will still take into account your feedback for this PR.

signing process even faster for your customers, partners, and employees.

To create and customize fields, activate the :ref:`developer mode <developer-mode>`. Then head to
:menuselection:`Documents --> Configuration --> Field Types` and click on *Create*.

After giving your new field a name, select one of the six *Types* available:

- **Signature**: the user is prompted to enter their signature either by drawing it, automatically
  generating it based on their name, or loading a local file (usually an image). Each subsequent
  *Signature* field then reuses the data entered in the first field.
- **Initial**: the user is prompted to enter their initials, similarly to the *Signature* field.
- **Text**: the user enters any text on a single line.
- **Multiline Text**: the user enters any text on multiple lines.
- **Checkbox**: the user can tick a box to mark their approval or consent, for example.
- **Selection**: the user chooses a single option between a series of options.

Next, you have the option to auto-complete fields for users based on their Odoo profile information
by using *Automatic Partner Field*. To this end, the *Name*, *Email*, *Phone*, and *Company* fields

> are preconfigured in your database.

Author: More can be configured.

> are preconfigured in your database.

Author: More can be configured.

.. note::
   Users can freely edit auto-completed fields.

.. image:: sign/field-example.png
   :align: center
   :alt: Company field example in Odoo Sign

You can then change the size of the field by editing the *Default Width* and *Default Height*,
expressed as a ratio of 1, 1 equalling the full-page's width or height. By default, the width of new
fields you create is set to 15% (0.150) of full-page's width, while their height is set to 1.5%
(0.015) of a full-page's height.

> Next, you should write a *Tip*. Tips are displayed inside arrows on the right side of the user's

Reviewer 3: ```suggestion
Next, write a *Tip*. Tips are displayed inside arrows on the right-hand side of the user's
```
Given that you tips are required, it makes more sense in my head to just write a tip since it's a step that they can't skip. Feel free to ignore though

screen during the signing process. You can also use a *Placeholder* text which is displayed inside

> the field before it is completed.

Reviewer 3: Wouldn't it just be "placeholder text" or is the text a specific type of field or tip? The way it's written now is a bit awkward.

Author: The field is named *Placeholder*
